### ISSUE TITLE
Impossible to update custom themes once added

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#13222]
It's an array of objects, and the URL field just appends to that array. In the UI we dedupe on name, which means we only ever get the first option. We should dedupe on name at add time.

**Comment by ellietheoracle** [comment]
updating the source should also update the theme ? shouldn't it just re-download the theme whenever riot is loading?

**Comment by ellietheoracle** [comment]
shouldn't it at least be an option?

**Comment by t3chguy** [comment]
> We should dedupe on name at add time.

Yes, this issue specifies that if you enter the URL again it should update the theme.

**Comment by ellietheoracle** [comment]
well i meant an option to reload the theme on start, it's much more convenient

**Comment by ellietheoracle** [comment]
some people don't care about telemetry

**Comment by t3chguy** [comment]
> some people don't care about telemetry

Sure but pretty much everything in this app does, for privacy.

**Comment by ellietheoracle** [comment]
fair enough ig, maybe a reload button then?

**Comment by turt2live** [comment]
The URL is only used to get the JSON object, after which the URL is never seen again.

**Comment by ellietheoracle** [comment]
would there be an issue with storing the url  with the json?

**Comment by turt2live** [comment]
probably not, but it makes the UI a whole lot more complicated and the data storage harder to maintain. It was excluded for largely reasons of not providing that much gain - instead, users should be able to just update it manually.
